src/pudl/analysis/plant_parts_eia.py

```diff
@@ -1768,3 +1491,96 @@ def assign_record_id_eia(test_df: pd.DataFrame, plant_part_col: str = "plant_par
         )
     test_df_ids = pd.concat(dfs)
     return test_df_ids
+
+
+def match_to_single_plant_part(
+    multi_gran_df: pd.DataFrame,
+    ppl: pd.DataFrame,
+    part_name: Literal[tuple(PLANT_PARTS.keys())] = "plant_gen",
```

Comment:
Ugh is there a better way to do this? I want to restrict the values of `part_name` to be the keys in `PLANT_PARTS`.